File: Frontend/src/app/models/RouteEvaluation.ts
/**
 * This Model represents a route evaluation given by an user
 */
export class RouteEvaluation {

    /**
     * Construct's an instance of {@link RouteEvaluation}
     * 
     * @param evaluation route's evaluation
     * @param userName user name of the user that evaluated the route
     * @param routeId route's id that was evaluated
     */
    constructor(
        public evaluation: Number,
        public userName: string,
        public routeId: Number
    ) { }
}

File: Backend/Data/Models/RouteEvaluation.cs

namespace Backend.Data.Models
{
    public class RouteEvaluation
    {
        public RouteEvaluation(long evaluation, string userName, long routeId)
        {
            Evaluation = evaluation;
            UserName = userName;
            User = null!;
            RouteId = routeId;
            Route = null!;
        }

        public long Evaluation { get; set; }


        //Navigations Properties 


        public string UserName { get; set; }//foreignKey
        public virtual User User { get; set; }//foreignKey

        public long RouteId { get; set; }//foreignKey   
        public virtual Route Route { get; set; }//foreignKey           


    }
}